starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java

```diff
@@ -111,4 +111,40 @@ public void delete() {
     Assert.assertTrue(outInv.isPresent());
     Assert.assertEquals(this.mongoTemplate.estimatedCount(Inventory.class), 0);
   }
+
+  /**
+   * Test retrieve method.
+   */
+  @Test
+  public void retrieve() {
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    Inventory outInv = this.inventoryDAO.create(inventory);
+    Assert.assertEquals(this.mongoTemplate.estimatedCount(Inventory.class), 1);
+
+    Optional<Inventory> outInv2 = this.inventoryDAO.retrieve(outInv.getId());
+    Assert.assertTrue(outInv2.isPresent());
+    Assert.assertEquals(outInv2.get(), outInv);
+  }
+
+  /**
+   * Test update method.
+   */
+  @Test
+  public void update() {
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    Inventory outInv = this.inventoryDAO.create(inventory);
+    Assert.assertEquals(this.mongoTemplate.estimatedCount(Inventory.class), 1);
+
+    Inventory newInventory = inventory;
+    newInventory.setName("UpdatedInv");
+    Optional<Inventory> outInv2 = this.inventoryDAO.update(outInv.getId(), newInventory);
+    
+    Assert.assertTrue(outInv2.isPresent());
+    Assert.assertNotSame(outInv2.get(), outInv);
+    Assert.assertEquals(outInv2.get(), newInventory);
```

Comment:
outInv2 may not always be the same as newInventory, in this case they are, but you could have used a different inventory with a unique id to update the original with, so in that case the ids wouldn't be the same.